src/cid.rs

```diff
@@ -263,7 +298,7 @@ where
     }
 }
 
-impl<C, H> From<CidGeneric<C, H>> for Vec<u8>
+impl<C, H> From<CidGeneric<C, H>> for Box<[u8]>
```

Comment:
Is there any utility in keeping the `Vec<u8>` impl around in addition to `Box<[u8]>` for more flexibility?

Reply:
Probably not, since we want it to be a fixed size :thinking:

Reply:
I wondered the same.

Reply:
If people want more flexibility they should use `copy_to_slice`

Reply:
FWIW I like the `Box<[u8]>` here as it signals to the reader that the value there is of fixed size and they should create a new Boxed value instead of mutating a `Vec`

Reply:
I disagree, it is usefull to get things from a `Vec`, most programs do not work with boxed slices.